astroquery/mast/missions.py

```diff
@@ -0,0 +1,201 @@
+        params = {'target': [f"{coordinates.ra.deg} {coordinates.dec.deg}"],
+                  'radius': radius.arcmin,
+                  'radius_units': 'arcminutes'}
+
+
+        # adding additional user specified parameters
+        for prop, value in kwargs.items():
+            params[prop] = value
+
+        return self._service_api_connection.service_request_async(self.service, params, use_json=True)
+
+    @class_or_instance
+    def query_criteria_async(self, **criteria):
+        """
+        Given an set of filters, returns a list of catalog entries.
+
+        Parameters
+        ----------
+        **criteria
+            Criteria to apply. At least one non-positional criteria must be supplied.
+            Valid criteria are coordinates, objectname, radius (as in `query_region` and `query_object`),
+            and all fields listed in the column documentation for the mission being queried.
+
+        Returns
+        -------
+        response : list of `~requests.Response`
+        """
+
+        # Seperating any position info from the rest of the filters
+        coordinates = criteria.pop('coordinates', None)
+        objectname = criteria.pop('objectname', None)
+        radius = criteria.pop('radius', 0.2*u.deg)
+
+        if objectname or coordinates:
+            coordinates = utils.parse_input_location(coordinates, objectname)
+
+        # if radius is just a number we assume degrees
+        radius = coord.Angle(radius, u.arcmin)
+
+        # build query
+        params = {}
+        if coordinates:
+            params["target"] = [f"{coordinates.ra.deg} {coordinates.dec.deg}"]
+            params["radius"] = radius.arcmin
+            params["radius_units"] = 'arcminutes'
+
+        if not self._service_api_connection.check_catalogs_criteria_params(criteria):
+            raise InvalidQueryError("At least one non-positional criterion must be supplied.")
+
+        for prop, value in criteria.items():
+            params[prop] = value
+
+        return self._service_api_connection.service_request_async(self.service, params, use_json=True)
+
+    @class_or_instance
+    def query_object_async(self, objectname, radius=3*u.arcmin, **kwargs):
+        """
+        Given an object name, returns a list of catalog entries.
+
+        Parameters
+        ----------
+        objectname : str
+            The name of the target around which to search.
+        radius : str or `~astropy.units.Quantity` object, optional
+            Default 3 arcmin.
+            The string must be parsable by `~astropy.coordinates.Angle`.
+            The appropriate `~astropy.units.Quantity` object from
+            `~astropy.units` may also be used. Defaults to 0.2 deg.
+        **kwargs
+            Mission-specific keyword args.
+            These can be found in the `service documentation <https://mast.stsci.edu/api/v0/_services.html>`__.
+            for specific catalogs. For example one can specify the magtype for an HSC search.
```

Comment:
Same as in `query_region` pull out mission as a keyword specifically.

Reply:
still unaddressed

Reply:
mission is at class level now